src/git-version.cr

```diff
@@ -139,7 +136,7 @@ module GitVersion
       major = false
       get_commits_since(previous_tag).each do |c|
         commit = c.downcase
-        if commit.includes?(MAJOR_BUMP_COMMENT)
+        if /#{@major_identifier}/.match(commit)
```

Comment:
The change to regex will likely break the current regular usage in case the identifier contains Regex chars like `.()[]-`.

Reply:
Fixed, users can now configure whether regex or string matching is used

Reply:
Except the unlikely case that their match starts and ends with a slash